Data Source: [Kaggle](!https://www.kaggle.com/datasets/joelljungstrom/128k-airline-reviews)

In [20]:
import pandas as pd

# ar_first_half = ar.loc[:round(ar.shape[0]/2)]
# ar_second_half = ar.loc[round(ar.shape[0]/2)+1:]
# ar_first_half.to_csv("AirlineReviews1.csv")
# ar_second_half.to_csv("AirlineReviews2.csv")

# Read seperated data sources
ar1 = pd.read_csv("AirlineReviews1.csv")
ar2 = pd.read_csv("AirlineReviews2.csv")
# Union datasets
ar = pd.concat([ar1, ar2], axis=0)
# Drop unnecassery column
ar = ar.drop(ar.columns[0], axis=1)

In [23]:
# Top 10 occurences in database
ar.groupby('AirlineName').size().reset_index(name='count').sort_values(by='count', ascending=False).head(10)

AirlineName  count
88   American Airlines   5509
437    Spirit Airlines   4829
500    United Airlines   4506
135    British Airways   3541
215  Frontier Airlines   3136
176    Delta Air Lines   2770
492   Turkish Airlines   2298
192           Emirates   2254
301          Lufthansa   2203
380      Qatar Airways   2196

In [55]:
ar.

0    Moroni to Moheli. Turned out to be a pretty de...
0    I’m from Virginia and this was my first time f...
Name: Review, dtype: object

In [33]:
# Group by 'AirlineName' and count occurrences, then sort in descending order
grouped_counts = ar.groupby('AirlineName').size().sort_values(ascending=False)

# Get top 10 airlines
top_10_airlines = grouped_counts.head(10).index

# Filter the DataFrame to only include rows for the top 10 airlines
top_10_df = ar[ar['AirlineName'].isin(top_10_airlines)]

# Select columns ending with 'Rating' or 'OverallScore' and calculate the average for each airline
average_ratings = top_10_df.filter(like='Rating').groupby(top_10_df['AirlineName']).mean()

average_ratings

EntertainmentRating  FoodRating  GroundServiceRating  \
AirlineName                                                               
American Airlines             1.278998    1.392993             1.488110   
British Airways               1.852866    2.462016             2.158995   
Delta Air Lines               2.025993    2.061011             1.944404   
Emirates                      3.670364    3.019077             2.126442   
Frontier Airlines             0.507015    0.680166             1.162309   
Lufthansa                     2.193827    2.820245             2.079437   
Qatar Airways                 3.748179    3.808743             2.999545   
Spirit Airlines               0.500725    0.753572             1.271899   
Turkish Airlines              2.590078    2.884247             1.693211   
United Airlines               1.511984    1.560586             1.579893   

                   SeatComfortRating  ServiceRating  ValueRating  WifiRating  
AirlineName                                                                   
American Airlines           1.969504       2.123434     1.739336    0.811036  
British Airways             2.811635       3.166902     2.724372    0.300762  
Delta Air Lines             2.567148       2.827437     2.359928    1.066065  
Emirates                    3.311890       3.033718     3.136202    1.274623  
Frontier Airlines           1.458227       1.702168     1.515306    0.302934  
Lufthansa                   2.984113       3.414435     3.039946    0.639582  
Qatar Airways               3.934426       4.205829     3.984973    1.533698  
Spirit Airlines             1.498033       1.835991     1.675502    0.303375  
Turkish Airlines            2.752393       2.828547     2.778068    0.750653  
United Airlines             2.080337       2.312472     1.915224    0.814913

In [35]:
top_10_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 33242 entries, 20202 to 62537
Data columns (total 22 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Aircraft             9229 non-null   object 
 1   AirlineName          33242 non-null  object 
 2   CabinType            32483 non-null  object 
 3   DateFlown            27526 non-null  object 
 4   DatePub              33242 non-null  object 
 5   EntertainmentRating  33242 non-null  int64  
 6   FoodRating           33242 non-null  int64  
 7   GroundServiceRating  33242 non-null  int64  
 8   OriginCountry        33229 non-null  object 
 9   OverallScore         32942 non-null  float64
 10  Recommended          33242 non-null  object 
 11  Review               32965 non-null  object 
 12  Route                27430 non-null  object 
 13  SeatComfortRating    33242 non-null  int64  
 14  ServiceRating        33242 non-null  int64  
 15  Slug                 33242 non-null  

| AirlineName      | EntertainmentRating | FoodRating | GroundServiceRating | OverallScore | SeatComfortRating | ServiceRating | ValueRating | WifiRating |
|------------------|---------------------|------------|---------------------|--------------|-------------------|---------------|-------------|------------|
| American Airlines| 1.278998            | 1.392993   | 1.48811             | 2.436773     | 1.969504          | 2.123434      | 1.739336    | 0.811036   |
| British Airways  | 1.852866            | 2.462016   | 2.158995            | 4.8125       | 2.811635          | 3.166902      | 2.724372    | 0.300762   |
| Delta Air Lines  | 2.025993            | 2.061011   | 1.944404            | 3.964312     | 2.567148          | 2.827437      | 2.359928    | 1.066065   |
| Emirates         | 3.670364            | 3.019077   | 2.126442            | 5.410564     | 3.31189           | 3.033718      | 3.136202    | 1.274623   |
| Frontier Airlines| 0.507015            | 0.680166   | 1.162309            | 1.846932     | 1.458227          | 1.702168      | 1.515306    | 0.302934   |
| Lufthansa        | 2.193827            | 2.820245   | 2.079437            | 5.490241     | 2.984113          | 3.414435      | 3.039946    | 0.639582   |
| Qatar Airways    | 3.748179            | 3.808743   | 2.999545            | 7.392711     | 3.934426          | 4.205829      | 3.984973    | 1.533698   |
| Spirit Airlines  | 0.500725            | 0.753572   | 1.271899            | 2.259729     | 1.498033          | 1.835991      | 1.675502    | 0.303375   |
| Turkish Airlines | 2.590078            | 2.884247   | 1.693211            | 4.417068     | 2.752393          | 2.828547      | 2.778068    | 0.750653   |
| United Airlines  | 1.511984            | 1.560586   | 1.579893            | 2.833962     | 2.080337          | 2.312472      | 1.915224    | 0.814913   |


In [39]:
# Select columns with float data type (presumably ratings columns)
rating_columns = top_10_df.select_dtypes(include=['float','int64']).columns

# Calculate the average for each airline
average_ratings = top_10_df[rating_columns].groupby(top_10_df['AirlineName']).mean()
average_ratings


EntertainmentRating  FoodRating  GroundServiceRating  \
AirlineName                                                               
American Airlines             1.278998    1.392993             1.488110   
British Airways               1.852866    2.462016             2.158995   
Delta Air Lines               2.025993    2.061011             1.944404   
Emirates                      3.670364    3.019077             2.126442   
Frontier Airlines             0.507015    0.680166             1.162309   
Lufthansa                     2.193827    2.820245             2.079437   
Qatar Airways                 3.748179    3.808743             2.999545   
Spirit Airlines               0.500725    0.753572             1.271899   
Turkish Airlines              2.590078    2.884247             1.693211   
United Airlines               1.511984    1.560586             1.579893   

                   OverallScore  SeatComfortRating  ServiceRating  \
AirlineName                                                         
American Airlines      2.436773           1.969504       2.123434   
British Airways        4.812500           2.811635       3.166902   
Delta Air Lines        3.964312           2.567148       2.827437   
Emirates               5.410564           3.311890       3.033718   
Frontier Airlines      1.846932           1.458227       1.702168   
Lufthansa              5.490241           2.984113       3.414435   
Qatar Airways          7.392711           3.934426       4.205829   
Spirit Airlines        2.259729           1.498033       1.835991   
Turkish Airlines       4.417068           2.752393       2.828547   
United Airlines        2.833962           2.080337       2.312472   

                   ValueRating  WifiRating  
AirlineName                                 
American Airlines     1.739336    0.811036  
British Airways       2.724372    0.300762  
Delta Air Lines       2.359928    1.066065  
Emirates              3.136202    1.274623  
Frontier Airlines     1.515306    0.302934  
Lufthansa             3.039946    0.639582  
Qatar Airways         3.984973    1.533698  
Spirit Airlines       1.675502    0.303375  
Turkish Airlines      2.778068    0.750653  
United Airlines       1.915224    0.814913

In [41]:
ar.Route.value_counts()

Route
Sydney to Melbourne              149
Melbourne to Sydney              144
Toronto to Vancouver             104
Vancouver to Toronto             104
Bangkok to Phuket                 73
                                ... 
Cologne to Edinburgh               1
Cologne to Hamburg                 1
Düsseldorf to Geneva               1
Phuket to Hamburg via Cologne      1
Paris Beauvais to Riga             1
Name: count, Length: 51401, dtype: int64

In [49]:
# Assuming your data is stored in a pandas DataFrame called 'df' with a column named 'Date'
ar['Date'] = pd.to_datetime(ar['DateFlown'])

# If the format is not recognized, you can specify the format explicitly
# For example, if the format is 'Month Year'
ar['Date'] = pd.to_datetime(ar['Date'], format='%B %Y')

# Displaying the converted dates
ar['Date'].describe()

C:\Users\husee\AppData\Local\Temp\ipykernel_13836\2011831643.py:2: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  ar['Date'] = pd.to_datetime(ar['DateFlown'])


count                            90993
mean     2018-12-27 08:56:00.363984128
min                2012-04-01 00:00:00
25%                2017-01-01 00:00:00
50%                2018-11-01 00:00:00
75%                2020-11-01 00:00:00
max                2023-05-01 00:00:00
Name: Date, dtype: object